Source/Models/Proof.swift

```diff
@@ -10,9 +10,24 @@ public struct Proof {
     
     let nodeHash: NodeHash
     let status: Status
+    let metadata: [String: Any]?
     
     static func make(from chainPointProof: ChainpointProofResponse, with nodeHash: NodeHash) -> Proof {
-        return Proof(nodeHash: nodeHash,
-                     status: .partial) // TODO: Daniel Metzing - what is the status? Discuss: What should be saved from the response?
+        
+        var metadata: [String: Any]?
+        
+        do {
+            let jsonData = try JSONEncoder().encode(chainPointProof)
+            let decoded = try JSONSerialization.jsonObject(with: jsonData, options: [])
+            
+            if let dictFromJSON = decoded as? [String: Any] {
+                metadata = dictFromJSON
+            }
+        } catch {
+            print("Proof.make -> json error: \(error)")
+        }
+
+        let proof = Proof(nodeHash: nodeHash, status: .partial, metadata: metadata)
```

Comment:
This metadata is alway there no? Shall we not make the whole function throwing, instead of having the metadata as optional on the proof?

Reply:
If I see correctly the metadata could be nil, beacuse if the user dont wait some second before pull the Proof from the server almost every metadata will be nil

Reply:
This could be a response where the metadata is still empty
```
[
  {
    "hash_id_node": "20a38230-afb8-11e8-8282-01626f373351",
    "proof": null,
    "anchors_complete": []
  }
]
```